lib/twingly/url.rb

```diff
@@ -6,7 +6,19 @@
 require_relative "url/error"
 require_relative "version"
 
-PublicSuffix::List.private_domains = false
+PublicSuffix::List.default = PublicSuffix::List.parse(
+  File.read(PublicSuffix::List::DEFAULT_LIST_PATH), private_domains: false)
+
+# Add ASCII form of all internationalized domain names to the public suffix list
+PublicSuffix::List.default.
+  map { |rule| Addressable::IDNA.to_ascii(rule.value) }.
```

Comment:
I tried manually with

```diff
diff --git a/lib/twingly/url.rb b/lib/twingly/url.rb
index 4193abd..9dd4581 100644
--- a/lib/twingly/url.rb
+++ b/lib/twingly/url.rb
@@ -1,5 +1,5 @@
 require "addressable/uri"
-require "addressable/idna/native"
+# require "addressable/idna/native"
 require "public_suffix"

 require_relative "url/null_url"
@@ -29,7 +29,7 @@ module Twingly
     ERRORS_TO_EXTEND = [
       Addressable::URI::InvalidURIError,
       PublicSuffix::DomainInvalid,
-      IDN::Idna::IdnaError,
+      # IDN::Idna::IdnaError,
     ]

     private_constant :ACCEPTED_SCHEMES, :ENDS_WITH_SLASH, :ERRORS_TO_EXTEND
diff --git a/twingly-url.gemspec b/twingly-url.gemspec
index f06d0d9..89f14cd 100644
--- a/twingly-url.gemspec
+++ b/twingly-url.gemspec
@@ -14,7 +14,7 @@ Gem::Specification.new do |s|

   s.add_dependency "addressable", "~> 2"
   s.add_dependency "public_suffix", "~> 2"
-  s.add_dependency "idn-ruby", "~> 0.1"
+  # s.add_dependency "idn-ruby", "~> 0.1"

   s.add_development_dependency "rake", "~> 10"
   s.add_development_dependency "rspec", "~> 3"
```

```
~/twingly/ruby/twingly-url public_suffix-2*
$ bundle show | grep idn-ruby
```

and all specs pass, but I'm not sure it says much, as we just have a few IDNs in our tests cases. But I think it's okay. :)